src/Core__Int.resi

```diff
@@ -0,0 +1,273 @@
+
+/**
+`toPrecision(n, ~digits)` return a `string` representing the giver value with
+precision. `digits` specifies the number of significant digits. See [`Number.toPrecision`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/toPrecision) on MDN.
+
+## Examples
+
+```rescript
+Int.toPrecision(100, ~digits=2) // "1.0e+2"
+Int.toPrecision(1) // "1.0"
+```
+
+## Expections
+
+`RangeError`:
+
+- If `~digits` is not between 1 and 100 (inclusive). Implementations are allowed
+to support larger and smaller values as well. ECMA-262 only requires a precision
+of up to 21 significant digits.
+
+*/
+@send
+external toPrecisionWithPrecision: (int, ~digits: int) => string = "toPrecision"
+
+/**
+`toString(n)` return a `string` representing the given value.
+See [`Number.toString`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/toString)
+on MDN.
+
+## Examples
+
+```rescript
+Int.toString(1000) // "1000"
+Int.toString(-1000) // "-1000"
+```
+*/
+@send
+external toString: int => string = "toString"
+
+/**
+`toStringWithRadix(n, ~radix)` return a `string` representing the given value.
+`~radix` specifies the radix base to use for the formatted number.
+See [`Number.toString`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/toString)
+on MDN.
+
+## Examples
+
+```rescript
+Int.toString(6, ~radix=2) // "110"
+Int.toString(3735928559, ~radix=16) // "deadbeef"
+Int.toStringWithRadix(123456, ~radix=36) // "2n9c"
+```
+
+## Exceptions
+
+`RangeError`:
+
+- if `~radix` is less than 2 or greater than 36.
+*/
+@send
+external toStringWithRadix: (int, ~radix: int) => string = "toString"
+
+/**
+`toLocaleString(n)` return a `string` with language-sensitive representing the
+given value. See [`Number.toLocaleString`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/toLocaleString) on MDN.
+
+## Examples
+
+```rescript
+// If the application uses English as the default language
+Int.toLocaleString(1000) // "1,000"
+
+// If the application uses Portuguese Brazil as the default language
+Int.toLocaleString(1000) // "1.000"
+```
+*/
+@send
+external toLocaleString: int => string = "toLocaleString"
+
+/**
+`toFloat(n)` return a `float` representing the given value.
+
+## Examples
+
+```rescript
+Int.toFloat(100) == 100.
+Int.toFloat(2) == 2.0
+```
+*/
+external toFloat: int => float = "%identity"
+
+/**
+`fromFloat(n)` return an `int` representing the given value.
+
+## Examples
+
+```rescript
+Int.fromFloat(2.0) == 2
+Int.fromFloat(1.0) == 1
+```
+*/
+external fromFloat: float => int = "%intoffloat"
+
+/**
+`fromString(~radix?, v)` return an `option<int>` representing the given value
+`v`. `~radix` specifies the radix base to use for the formatted number.
+
+## Examples
+
+```rescript
+Int.fromString("0") == Some(0)
+Int.fromString("NaN") == None
+```
+*/
+let fromString: (~radix: int=?, 'a) => option<int>
```

Comment:
This function signature should not be `(string, ~radix: int=?) => option<int>`?

Reply:
Yeah that seems more logical. Although we should preserve the current order of the arguments.